src/testing/stability_fr_parameter_range.cpp

```diff
@@ -0,0 +1,354 @@
+    std::vector<double> grid_size(n_grids);
+    std::vector<double> soln_error(n_grids);
+    std::vector<double> soln_error_inf(n_grids);
+    unsigned int poly_degree = all_parameters_new.manufactured_convergence_study_param.degree_start;
+    dealii::ConvergenceTable convergence_table;
+    const unsigned int igrid_start = all_parameters_new.manufactured_convergence_study_param.initial_grid_size;
+    pcout << " igrid_start" << igrid_start << std::endl;
+    const unsigned int grid_degree = 1;
+
+    const unsigned int nb_c_value = all_parameters_new.number_ESFR_parameter_values;
+    const double c_min = all_parameters_new.ESFR_parameter_values_start;
+    const double c_max = all_parameters_new.ESFR_parameter_values_end;
+    const double log_c_min = std::log10(c_min);
+    const double log_c_max = std::log10(c_max);
+    std::vector<double> c_array(nb_c_value+1);
+
+    std::ofstream l2error_file("l2error.txt");
+    std::ofstream slope_file("slope_soln_err.txt");
+    std::ofstream c_value_file("c_value.txt");
+    std::ofstream cell_number_file("cell_number.txt");
+
+    for(unsigned int igrid = igrid_start; igrid<n_grids; igrid++){
+        cell_number_file << std::pow(2.0, igrid) << " ";
+    }
+    cell_number_file.close();
+
+    // Create log space array of c_value
+    for (unsigned int ic = 0; ic < nb_c_value; ic++) {
+        double log_c = log_c_min + (log_c_max - log_c_min) / (nb_c_value - 1) * ic;
+        c_array[ic] = std::pow(10.0, log_c);
+        c_value_file << c_array[ic] << std::endl;
+    }
+    // c_array[nb_c_value]=4.24e-7; // 0.186; 3.67e-3; 4.79e-5; 4.24e-7;
+    c_array[nb_c_value] = all_parameters_new.FR_user_specified_correction_parameter_value;
+    c_value_file << c_array[nb_c_value] << std::endl;
+    c_value_file.close();
+
+    // Loop over c_array to compute slope
+    for (unsigned int ic = 0; ic < nb_c_value+1; ic++) {
+        double c_value = c_array[ic];
+
+    for(unsigned int igrid = igrid_start; igrid<n_grids; igrid++){
+
+#if PHILIP_DIM==1 // dealii::parallel::distributed::Triangulation<dim> does not work for 1D
+        using Triangulation = dealii::Triangulation<dim>;
+        std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+            typename dealii::Triangulation<dim>::MeshSmoothing(
+                dealii::Triangulation<dim>::smoothing_on_refinement |
+                dealii::Triangulation<dim>::smoothing_on_coarsening));
+#else
+        using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+        std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+            MPI_COMM_WORLD,
+            typename dealii::Triangulation<dim>::MeshSmoothing(
+                dealii::Triangulation<dim>::smoothing_on_refinement |
+                dealii::Triangulation<dim>::smoothing_on_coarsening));
+#endif
+        //straight grid setup
+        dealii::GridGenerator::hyper_cube(*grid, left, right, true);
```

Comment:
You can generate the grid using `Grids::straight_periodic_cube`. See https://github.com/dougshidong/PHiLiP/blob/474544d87673c214cd74c9fe45559cf0aad6824f/src/flow_solver/flow_solver_cases/periodic_cube_flow.cpp#L60